src/components/custom-field-input-single-choice/custom-field-input-single-choice.jsx

```diff
@@ -68,23 +73,54 @@ const CustomFieldInputSingleChoice = forwardRef(function CustomFieldInputSingleC
     props.onChange({ target: selfRef.current });
   }, [value]);
 
+  useEffect(() => {
+    const getChoices = async () => {
+      await execute(`${API_ROOT}/custom_field_choices?for_custom_fields=${props.id}`)
+        .then(({ json, mounted }) => {
+          if (mounted) {
+            const mungedChoices = json.results.map((result) => {
+              return {
+                id: result.id,
+                label: json[result.key][result.id].label,
+              };
+            });
+
+            setChoices(mungedChoices);
+            setListOptionRefs(mungedChoices.map(() => createRef()));
```

Comment:
This is interesting. What do you think about simplifying our state management by dropping `setListOptionRefs` as part of state?

It is a calculated piece of state that will never be anything that is not derived from the length of loaded choices.

Instead, do this piece of logic at the start of the function:

(I'll add the suggestion at the relevant line)